solidity/contracts/hooks/ArbitrumMessageHook.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+import "forge-std/console.sol";
+
+// ============ Internal Imports ============
+import {IArbitrumMessageHook} from "../interfaces/hooks/IArbitrumMessageHook.sol";
+import {ArbitrumISM} from "../isms/native/ArbitrumISM.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+
+// ============ External Imports ============
+import {IInbox} from "@arbitrum/nitro-contracts/src/bridge/IInbox.sol";
+
+/**
+ * @title ArbitrumMessageHook
+ * @notice Message hook to inform the Arbitrum ISM of messages published through
+ * the native Arbitrum bridge.
+ */
+contract ArbitrumMessageHook is IArbitrumMessageHook {
+    // ============ Constants ============
+
+    // Domain of chain on which the arbitrum ISM is deployed
+    uint32 public immutable destinationDomain;
+    // Arbitrum's inbox used to send messages from L1 -> L2
+    IInbox public immutable inbox;
+
+    uint128 internal constant MAX_GAS_LIMIT = 100_000;
+    uint128 internal constant MAX_GAS_PRICE = 1e9;
```

Comment:
How did you land on this number?

Reply:
receiveFromHook costs 25k gas x 4

Reply:
Gas price?

Reply:
https://arbiscan.io/address/0x000000000000000000000000000000000000006C#readContract arbGasInfo.getMinimumGasFee(). gives you 1e8 or 0.1 gwei and I made it 10x that conservatively. Maybe we can the app to specify this?